src/frontends/onnx/frontend/src/op/com.microsoft/range.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/com.microsoft/range.hpp"
+
+#include <memory>
+
+#include "default_opset.hpp"
+
+OPENVINO_SUPPRESS_DEPRECATED_START
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+OutputVector range(const Node& node) {
+    const Output<ngraph::Node> start{node.get_ng_inputs().at(0)};
+    const Output<ngraph::Node> stop{node.get_ng_inputs().at(1)};
+    const Output<ngraph::Node> step{node.get_ng_inputs().at(2)};
+    return {std::make_shared<default_opset::Range>(start, stop, step, start.get_element_type())};
```

Comment:
[Refer here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/pad.cpp#L66-L77), you can use length of inputs to infer if `delta` has been provided or not.